shims/hdi40/driver/src/main/java/org/pentaho/hadoop/shim/hdi/format/HDIFormatShim.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ *
+ * Pentaho Big Data
+ *
+ * Copyright (C) 2002-2021 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+package org.pentaho.hadoop.shim.hdi.format;
+
+import org.pentaho.hadoop.shim.api.cluster.NamedCluster;
+import org.pentaho.hadoop.shim.api.format.IPentahoAvroInputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoAvroOutputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoInputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoOrcInputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoOrcOutputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoOutputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoParquetInputFormat;
+import org.pentaho.hadoop.shim.api.format.IPentahoParquetOutputFormat;
+import org.pentaho.hadoop.shim.common.CommonFormatShim;
+import org.pentaho.hadoop.shim.common.format.avro.PentahoAvroInputFormat;
+import org.pentaho.hadoop.shim.common.format.avro.PentahoAvroOutputFormat;
+import org.pentaho.hadoop.shim.hdi.format.orc.HDIOrcInputFormat;
+import org.pentaho.hadoop.shim.hdi.format.orc.HDIOrcOutputFormat;
+import org.pentaho.hadoop.shim.hdi.format.parquet.HDIApacheInputFormat;
+import org.pentaho.hadoop.shim.hdi.format.parquet.HDIApacheOutputFormat;
+
+public class HDIFormatShim extends CommonFormatShim {
```

Comment:
we've had to introduce a separate format + implementation for Orc and Parquet as opposed to the common format class for HDInsights, as we have additional steps to pass in the correct file system for the reader/writer and also for the makequalified funcationality as the HDInsights cluster storage pattern is different as opposed to other shims.

Ex: 

> <property>
      <name>fs.defaultFS</name>
      <value>wasb://wasb-secure-cluster-2021-05-17t09-37-48-203z@wasbclustersehdistorage.blob.core.windows.net</value>
      <final>true</final>
    </property>

Around 80% of code is the same as that in CommonShim classes however, we cannot extend the classes unless we make few of the fields protected under common-fragment repo to allow to be used under shim project and scope of change would be larger and extent to all other shims. With this, the change is limited to only HDInsights shim with code duplication being the only drawback.